app/ports/views.py

```diff
@@ -240,10 +279,69 @@ def stats(request):
         'last_week': last_week_unique,
         'users_by_month': users_by_month,
         'os_distribution': os_distribution,
-        'port_distribution': port_distribution,
         'macports_distribution': macports_distribution,
         'xcode_distribution': xcode_distribution,
-        'req_port_distribution': req_port_distribution
+        'days': int(days),
+        'allowed_days': allowed_days
+    })
+
+
+def stats_port_installations(request):
+    days = int(request.GET.get('days', 30))
+    first = request.GET.get('first', '-total_count')
+    second = request.GET.get('second', '-req_count')
+    third = request.GET.get('third', 'port')
+    values = [first, second, third]
+    acceptable_values = ['port', '-port', 'total_count', '-total_count', '-req_count', 'req_count']
+    allowed_days = [7, 30, 90, 180, 365]
+
+    if days not in allowed_days:
+        return HttpResponse("'days' key must have an integer value from the list: {}".format(allowed_days))
+
+    for i in values:
+        if i not in acceptable_values:
+            return HttpResponse("'{}' is an invalid value to sort. Allowed values: {}".format(acceptable_values))
```

Comment:
I think you missed to pass `i` in the `.format()`.

Reply:
Oh, thank you very much.